streams/src/main/java/org/apache/kafka/streams/processor/internals/AssignedStreamsTasks.java

```diff
@@ -46,6 +45,52 @@ public StreamTask restoringTaskFor(final TopicPartition partition) {
         return restoringByPartition.get(partition);
     }
 
+    @Override
+    List<StreamTask> allTasks() {
+        final List<StreamTask> tasks = super.allTasks();
+        tasks.addAll(restoring.values());
+        return tasks;
+    }
+
+    @Override
+    Set<TaskId> allAssignedTaskIds() {
+        final Set<TaskId> taskIds = super.allAssignedTaskIds();
+        taskIds.addAll(restoring.keySet());
+        return taskIds;
+    }
+
+    @Override
+    boolean allTasksRunning() {
+        return super.allTasksRunning() && restoring.isEmpty();
+    }
+
+    RuntimeException closeAllRestoringTasks() {
+        RuntimeException exception = null;
+
+        log.trace("Close restoring stream task {}", restoring.keySet());
+        final Iterator<StreamTask> restoringTaskIterator = restoring.values().iterator();
+        while (restoringTaskIterator.hasNext()) {
+            final StreamTask task = restoringTaskIterator.next();
+            log.debug("Closing restoring and not re-assigned task {}", task.id());
```

Comment:
Sorry for my denseness... Why are these "not re-assigned"? They're part of a data structure called "assigned tasks", which seems to imply that they are assigned.

Reply:
Terminology storm coming :)

The assigned tasks contains the following non-overlapping sets (see my other comment as well for details): created, restoring, running, suspended. Normally `created` should be empty since once a task is created it should move on transit to either restoring or running immediately. This function is called below for `suspendTasksAndState`. In which case we should:

1) `created` and `suspended`: no action, since it should be empty.
2) `running`: transit to `suspended`.
3) `restoring`: close immediately.

The comment itself is indeed misleading, I will change to `Closing restoring task`